chia/full_node/block_store.py

```diff
@@ -58,7 +58,11 @@ async def create(cls, db_wrapper: DBWrapper):
         await self.db.execute("CREATE INDEX IF NOT EXISTS height on block_records(height)")
 
         await self.db.execute("CREATE INDEX IF NOT EXISTS hh on block_records(header_hash)")
-        await self.db.execute("CREATE INDEX IF NOT EXISTS peak on block_records(is_peak)")
+
+        # this index was initially created and deployed without where condition "is_peak = 1"
+        # to replace the old index with the new one, uncomment the drop
+        # await self.db.execute("DROP INDEX IF EXISTS peak on block_records(is_peak)")
```

Comment:
I can't think of a reason to keep the old index. If we un-comment this line users will still be able to down-grade. There are no `INDEXED BY` clauses referring to this index by name, so downgrading will still just make sqlite use the new (smaller) index.

Unless you can think of a good reason to delay dropping this, I think we should do it right away in this patch.

Reply:
That's what I also assumed, but then you brought [this ](https://github.com/Chia-Network/chia-blockchain/pull/9226#issuecomment-966985939) up.

A possible solution could be to drop the old index and create the new partial index in a non-blocking manner, like establish a separate function that autonomously executes on startup and is used for such index maintenance tasks. 
But I have no python experience, so this no function I could add to this PR myself.

Reply:
Dropping the index might take ~25 seconds, and by not dropping it, we would save that startup time.

```
sqlite> drop index peak;
Run Time: real 24.339 user 0.102291 sys 0.296509
```

However, creating the new index may take as much as 1000 seconds.

```
sqlite> CREATE INDEX peak on block_records(is_peak) where is_peak = 1;
Run Time: real 1008.655 user 6.667830 sys 31.874013
```

So, creating the new index would dominate the cost in that case. Without dropping the old index, you also don't reap the benefits of the new index (afaict). In my measurements there's no speed improvement to use the smaller index. The benefit is a smaller database file. If you keep the old index, you don't get that improvement.

I suppose both dropping and creating the index could be done in a separate thread. The new one should probably be created first, before dropping the existing one. One risk with this is that spawning a thread and detaching it is risky, because we wouldn't be able to ensure it's done before shutting down. And just killing the thread while working on creating the index might be riskt (although, one would hope sqlite wouldn't create a half-baked index).

Reply:
I'm not sure it's worth introducing that complexity just to improve the experience on a one-time startup delay though..

Reply:
> So, creating the new index would dominate the cost in that case. Without dropping the old index, you also don't reap the benefits of the new index (afaict). In my measurements there's no speed improvement to use the smaller index. The benefit is a smaller database file. If you keep the old index, you don't get that improvement.

In theory, there should be a minimal speed improvement due to less index maintenance: no need to maintain old peak in index, but yes, the main benefit I see is the reduced size on the long run.

Reply:
> I'm not sure it's worth introducing that complexity just to improve the experience on a one-time startup delay though..

Agree, this would only make sense if such an improvement could be reused on every startup and not only for this specific index but also for all others.

So I'll just drop the index immediately and create new one on startup for this PR.